PatreonClient/PatreonHttpClient.cs

```diff
@@ -104,51 +105,30 @@ private IEnumerable<PatreonData> AggregateIncludes(string content)
             {
                 yield break;
             }
-
+            
             foreach (var el in included.EnumerateArray())
             {
                 var type = el.EnumerateObject().FirstOrDefault(x => x.Name == "type").Value.ToString();
-                if (type.Equals("campaign"))
-                {
-                    yield return
-                        JsonSerializer.Deserialize<PatreonData<Campaign, CampaignRelationships>>(el.ToString(), JsonSerializerOptions);
-                }
-                else if (type.Equals("member"))
-                {
-                    yield return JsonSerializer.Deserialize<PatreonData<Member, MemberRelationships>>(el.ToString(), JsonSerializerOptions);
-                }
-                else if (type.Equals("user") || type.Equals("creator") || type.Equals("patron"))
-                {
-                    yield return JsonSerializer.Deserialize<PatreonData<User, UserRelationships>>(el.ToString(), JsonSerializerOptions);
-                }
-                else if (type.Equals("tier"))
-                {
-                    var tier = JsonSerializer.Deserialize<PatreonData<Tier, TierRelationships>>(el.ToString(), JsonSerializerOptions);
-                    yield return tier;
-                }
-                else if (type.Equals("media"))
-                {
-                    yield return JsonSerializer.Deserialize<PatreonData<Media>>(el.ToString(), JsonSerializerOptions);
-                }
-                else if (type.Equals("benefit"))
-                {
-                    yield return JsonSerializer.Deserialize<PatreonData<Benefit, BenefitRelationships>>(el.ToString(), JsonSerializerOptions);
-                }
-                else if (type.Equals("deliverable"))
-                {
-                    yield return JsonSerializer.Deserialize<PatreonData<Deliverable, DeliverableRelationships>>(el.ToString(), JsonSerializerOptions);
-                }
-                else if (type.Equals("address"))
-                {
-                    yield return JsonSerializer.Deserialize<PatreonData<Address, AddressRelationships>>(el.ToString(), JsonSerializerOptions);
-                }
-                else if (type.Equals("goal"))
+
+                var decodeType = GetType().Assembly.GetTypes().First(e => e.GetCustomAttributes(typeof(ItemRelationshipAttribute), false) is ItemRelationshipAttribute[] a && a.Any(f => f.JsonName == type));
+
+                var attr = decodeType.GetCustomAttribute<ItemRelationshipAttribute>();
+                
+                if (attr.RelationshipType is null)
```

Comment:
I like the idea but something crazy is happening here.

I think it's better to have something like `ResponseType` on the ItemRelationshipAttribute and then we can have something like:
`return JsonSerializer.Deserialize(el.ToString(), attr.ResponseType)` I think for this you'll need to pass the type of the attributes in to the ItemRelationshipAttribute, so it'll be something like:

```
[AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
public class ItemRelationshipAttribute : Attribute
{
	public ItemRelationshipAttribute(Type attributeType, Type relationshipType = null)
	{
		// validation

		AttributeTypeName = attributeType.Name.ToLower();
		ResponseType = relationshipType == null
			? typeof(PatreonData<>).MakeGenericType(new[] { attributeType })
			: typeof(PatreonData<,>).MakeGenericType(new[] { attributeType, relationshipType });
	}
	
	public string AttributeTypeName { get; set; }
	public Type ResponseType
}
```
but probably still keep the json name due to creator etc..

Reply:
Solved in c240237